### ISSUE TITLE
Unit test target fails if BUILD_STATIC_LIBS is set to OFF

### ISSUE THREAD

**Issue by arkamar** [VectorCamp/vectorscan#202]
I am creating a package for Gentoo, I am configuring vectorscan with following command:
```
cmake -C /var/tmp/portage/dev-libs/vectorscan-5.4.11/work/vectorscan-vectorscan-5.4.11_build/gentoo_common_config.cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_BENCHMARKS=OFF -DBUILD_EXAMPLES=OFF -DBUILD_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=/var/tmp/portage/dev-libs/vectorscan-5.4.11/work/vectorscan-vectorscan-5.4.11_build/gentoo_toolchain.cmake /var/tmp/portage/dev-libs/vectorscan-5.4.11/work/vectorscan-vectorscan-5.4.11
```
It builds fine but when I try to run `unit` test target I get this error:
```
ninja: error: 'unit/unit-internal', needed by 'unit/CMakeFiles/unit', missing and no known rule to make it
```
I believe the reason for this is here:
https://github.com/VectorCamp/vectorscan/blob/d29730e1cb9daaa66bda63426cdce83505d2c809/unit/CMakeLists.txt#L61-L131
as I don't have `FAT_RUNTIME` nor `BUILD_STATIC_LIBS` set `ON`. I guess this issue is related to the commit 24ae1670d6bf72f2b01b3bd41d984d3dad66e68d which restricted `unit-internal` target for static libs only.

**Comment by markos** [comment]
Sorry for some reason I have missed this one. The reason for this is because linking this unit test with shared libs fails with undefined references. I do intend to fix this but didn't want to delay the release. But it's good that you created this bug report as a reminder.

**Comment by markos** [comment]
@arkamar sorry for being late in replying to this. So, the reason for not building the unit-internal with shared libs only is that the symbols are not exposed in the shared library. This is on purpose, if you want to run the unit-internal tests you need to build the static libs as the symbols are exposed there. I will close this issue now.

**Comment by arkamar** [comment]
I don't want to run `unit-internal` tests, I want to run just `unit` target with shared libs only but `unit-internal` is always inherited anyway, which is a bug in my opinion.

**Comment by markos** [comment]
ah, now I see the problem, apologies, I thought the problem was not being able to run unit-internal. yes, you're right

**Comment by arkamar** [comment]
No problem, and thank you for reopening.

**Comment by markos** [comment]
This should be fixed in #221

**Comment by arkamar** [comment]
The fix works for me, thanks!
